Firestore/core/src/firebase/firestore/nanopb/byte_string.h

```diff
@@ -80,10 +80,8 @@ class ByteString : public util::Comparable<ByteString> {
 
   ~ByteString();
 
-  ByteString& operator=(ByteString other) {
```

Comment:
Hmm, this is a little weird. IIUC, a function with `operator=(T)` signature should be considered a user-defined copy assignment operator[1], which should inhibit both the copy assignment operator and the move assignment operator from being implicitly defined. Even more confusingly, if the compiler were to implicitly define the copy assignment operator as `operator=(const T&)`, it would [create ambiguity](https://wandbox.org/permlink/9wDn0yThZnntwREQ). In other words, it seems like a compiler defect, and I'm not sure how it works (unless there's another accompanying defect that incorrectly resolves the ambiguity?).

[1] See [cppreference](https://en.cppreference.com/w/cpp/language/copy_assignment):
> A copy assignment operator of class `T` is a non-template non-static member function with the name `operator=` that takes exactly one parameter of type **`T`**, `T&`, `const T&`, `volatile T&`, or `const volatile T&`.

Reply:
I completely agree that this *should* work, but I can't argue with the fact that the leak detector is definitely seeing a leak and that this change definitely addresses it. I suspect more investigation in a more controlled environment could pin this down but spending that time now didn't seem worthwhile given the ready availability of a cheap, unobtrusive fix.